### ISSUE TITLE
No Controls 

### ISSUE THREAD

**Issue by AlexZ1402** [jaroschek/home-assistant-myuplink#83]
Hello,
I have a problem that only 2 adjustable values appear for me
premium account is available
I can read values via Swagger

how can I solve the problem?
anyone an idea?

Greetings
Alex
![Screenshot 2024-02-01 000639](https://github.com/jaroschek/home-assistant-myuplink/assets/119460011/ef71da49-4ff6-4b86-90cb-48fb3e893a40)

**Comment by jaroschek** [comment]
Hello, if you can read the values via swagger, then you should be able to see which parameter points are provides as `writable` in `/v2/devices/{deviceId}/points`. Only these parameters are translated into adjustable entities:

**Comment by AlexZ1402** [comment]
here as an example
set the temperature to 23°C
and the temperature is also set in the nibe heat pump without any problems





Curl

curl -X 'PATCH' \
  'https://api.myuplink.com/v2/devices/xxxxxxxxxxxxxxxxxx/points' \
  -H 'accept: text/plain' \
  -H 'Authorization: Bearer xxxxxxxxxxxxxxxxx
  -d '{
  "47398": "23"
}'



Server response
Code	Details
200	
Response body
Download
{
  "status": 200,
  "payload": {
    "47398": "modified"
  }
}

**Comment by AlexZ1402** [comment]
I think Nibe has a problem with the API, I have just connected to Google Home and the control is exactly the same, only these 2 can I control like in the Home Assistant

**Comment by willew** [comment]
> I think Nibe has a problem with the API

Agree, when I requested the device points using myUpLilnk Swagger (https://api.myuplink.com/swagger/index.html), the writeable device points are the same as in the HASS UI. For example the fan mode is read-only. I have Nibe F370.

But in myUplink Web UI, I can control the fan speed. When checking the network requests in the developer view, I can see that the fan speed is not requested with the documented API
```
Invoke-WebRequest -UseBasicParsing -Uri "https://internalapi.myuplink.com/v2/devices/emmy-r-191911-20240319-06605623192008-d8-47-8f-53-a0-30/points/47260" `
-Method "PUT" `

-ContentType "application/json" `
-Body "{`"value`":`"2`",`"unit`":`"`"}"
```
I could find any support service for the API, anyone seen?

**Comment by elden1337** [comment]
Control is a service call. Only for boost true false. Should work with turning on the switch manually also.
Mine is called: 

switch.[redacted]_increased_ventilation


![image](https://github.com/user-attachments/assets/e31d1ff9-a635-4958-8234-4b096a806052)

**Comment by arossmann** [comment]
> Control is a service call. Only for boost true false. Should work with turning on the switch manually also. Mine is called:
> 
> switch.[redacted]_increased_ventilation
> 
> ![image](https://private-user-images.githubusercontent.com/22585518/382501074-e31d1ff9-a635-4958-8234-4b096a806052.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2MjY0MzEsIm5iZiI6MTczMDYyNjEzMSwicGF0aCI6Ii8yMjU4NTUxOC8zODI1MDEwNzQtZTMxZDFmZjktYTYzNS00OTU4LTgyMzQtNGIwOTZhODA2MDUyLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwM1QwOTI4NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MjAyODI5YTA1OWEyNWRjNTY3YWEwNGE1OTc2NzVmNDZkMTNjN2ZjNTUzNGViNGI3M2ZlZWQ3ZWZlZGZlYjJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.HHu-lxEBAXB0kDSCNsPChHTuqzatU-UrDXaqN6DiNBc)

That was helping me, too
